autoload/airline/themes/dracula.vim

```diff
@@ -7,96 +7,84 @@
-
-let g:airline#themes#dracula#palette.normal.airline_warning = s:WARNING
-let g:airline#themes#dracula#palette.insert.airline_warning = s:WARNING
-let g:airline#themes#dracula#palette.visual.airline_warning = s:WARNING
-let g:airline#themes#dracula#palette.replace.airline_warning = s:WARNING
-
-let g:airline#themes#dracula#palette.normal.airline_error = s:ERROR
-let g:airline#themes#dracula#palette.insert.airline_error = s:ERROR
-let g:airline#themes#dracula#palette.visual.airline_error = s:ERROR
-let g:airline#themes#dracula#palette.replace.airline_error = s:ERROR
-
-" File modified and not saved
-let g:airline#themes#dracula#palette.normal_modified = airline#themes#generate_color_map(s:N1, s:N2, s:changed)
-let g:airline#themes#dracula#palette.insert_modified = airline#themes#generate_color_map(s:I1, s:I2, s:changed)
-let g:airline#themes#dracula#palette.replace_modified = airline#themes#generate_color_map(s:R1, s:R2, s:changed)
-let g:airline#themes#dracula#palette.visual_modified = airline#themes#generate_color_map(s:V1, s:V2, s:changed)
-
-" CtrlP
-if !get(g:, 'loaded_ctrlp', 0)
-  finish
+"}}}
+" Helpers: {{{
+
+func! s:clr(fg, bg, ...)
+  let l:fg = get(g:dracula_palette, a:fg)
+  let l:bg = get(g:dracula_palette, a:bg)
+  if ! type(l:fg) == 3
+    echoerr 'Color undefined in dracula_palette: ' . a:fg
+  elseif ! type(l:bg) == 3
+    echoerr 'Color undefined in dracula_palette: ' . a:fg
+  endif
+  return [ l:fg[0], l:bg[0], l:fg[1], l:bg[1] ] + filter(copy(a:000), 'type(v:val) == 1 && len(v:val) > 0')
+endfunc
+
+func! s:color_map(a, b, c, ...)
+  if a:0 == 0
+    return call('airline#themes#generate_color_map', [call('s:clr', a:a), call('s:clr', a:b), call('s:clr', a:c)])
+  else
+    return call('extend', [ call('airline#themes#generate_color_map', [call('s:clr', a:a), call('s:clr', a:b), call('s:clr', a:c)]) ] + a:000)
+  endif
+endfunc
+
+"}}}
+
+let g:airline#themes#dracula#palette = {
+\   'normal': s:color_map(
+\       ['bg', 'purple'],
+\       ['fg', 'comment'],
+\       ['fg', 'selection'],
+\       {
+\         'airline_warning': s:clr('bg', 'orange'),
+\         'airline_error': s:clr('bg', 'red'),
+\       },
+\     ),
+\   'insert': s:color_map(
+\       ['bg', 'green'],
+\       ['fg', 'comment'],
+\       ['fg', 'selection'],
+\       {
+\         'airline_warning': s:clr('bg', 'orange'),
+\         'airline_error': s:clr('bg', 'red'),
+\       },
+\     ),
+\   'replace': s:color_map(
+\       ['bg', 'orange'],
+\       ['fg', 'comment'],
+\       ['fg', 'selection'],
+\       {
+\         'airline_warning': s:clr('bg', 'orange'),
+\         'airline_error': s:clr('bg', 'red'),
+\       },
+\     ),
+\   'visual': s:color_map(
+\       ['bg', 'yellow'],
+\       ['fg', 'comment'],
+\       ['fg', 'selection'],
+\       {
+\         'airline_warning': s:clr('bg', 'orange'),
+\         'airline_error': s:clr('bg', 'red'),
+\       },
+\     ),
+\   'inactive': s:color_map(
```

Comment:
In inactive mode (i.e., with multiple windows open) the airline on regular buffers (not help, quickfix, or netrw with my testing) is unreadable in the left gutter. It appears the text becomes the same color as the grey background, becoming impossible to read.

Reply:
Actually, come to think of it, this was present before this commit... is this expected behavior ?

Reply:
Screenshot?

Reply:
Actually, ignore this. I found my problem. What happens is airline collapses the `mode` and `branch` sections, leaving the file name. I had been experimenting on a new file without a name. Editing a named file is the same as it was (readable but subtle and not in the way).